src/main/java/org/elasticsearch/index/query/BaseQueryBuilder.java

```diff
@@ -79,4 +79,10 @@ public Query toQuery(QueryParseContext parseContext) throws QueryParsingExceptio
     protected abstract String parserName();
 
     protected abstract void doXContent(XContentBuilder builder, Params params) throws IOException;
+
+    @Override
+    public void validate() {
+        // default impl does not validate, subclasses should override.
+        // TODO Remove once all queries support validation
+    };
```

Comment:
I don't think we need this if we move over query by query to extend this base class?